File: Programmers/2018_KAKAO_BLIND_RECREUITMENT/[3차] 방금그곡.java
class Solution {
    public String solution(String m, String[] musicinfos) {
        String answer = "";
        int max = 0;
            
        for (String musicinfo: musicinfos) {
            String[] info = musicinfo.split(",");
        
            String[] start = info[0].split(":");
            String[] end = info[1].split(":");
            int time = (Integer.parseInt(end[0]) - Integer.parseInt(start[0])) * 60 + Integer.parseInt(end[1]) - Integer.parseInt(start[1]);
            
            String replaced_code = replace_code(info[3]);
            String code = "";
            
            for (int i = 0; i < time / replaced_code.length(); i++)
                code += replaced_code;
            
            code += replaced_code.substring(0, time % replaced_code.length());
            
            if (code.contains(replace_code(m))) {
                if (max < time) {
                    max = time;
                    answer = info[2];
                }
            }
        }
        
        if (answer == "")
            answer = "(None)";
        
        return answer;
    }
    
    public String replace_code(String code) {
        code = code.replaceAll("C#", "c");
        code = code.replaceAll("D#", "d");
        code = code.replaceAll("F#", "f");
        code = code.replaceAll("G#", "g");
        code = code.replaceAll("A#", "a");

        return code;
    }
}

File: Programmers/2018_KAKAO_BLIND_RECREUITMENT/[3차] 방금그곡.py
def solution(m, musicinfos):
    answer = ''
    result = []
    
    for musicinfo in musicinfos:
        info = musicinfo.split(',')
        start = info[0].split(':')
        end = info[1].split(':')
        
        time = (int(end[0]) - int(start[0])) * 60 + int(end[1]) - int(start[1])

        code = replace_code(info[3])
        code = code * (time // len(code)) + code[:time%len(code)]
        
        if replace_code(m) in code:
            result.append([info[2], time])

    if len(result) == 0:
        return "(None)"
    
    else:
        result = sorted(result, key = lambda x: (-x[1]))
        return result[0][0]
    
def replace_code(code):
    code = code.replace('C#', 'c')
    code = code.replace('D#', 'd')
    code = code.replace('F#', 'f')
    code = code.replace('G#', 'g')
    code = code.replace('A#', 'a')
    
    return code